showcase/src/patterns/index.js

```diff
@@ -0,0 +1,195 @@
+    }
+
+    componentDidMount () {
+      const tlDuration = 300
+
+      const triangleBurst = new mojs.Burst({
+        parent: '#clap',
+        radius: { 50: 95 },
+        count: 5,
+        angle: 30,
+        children: {
+          shape: 'polygon',
+          radius: { 6: 0 },
+          scale: 1,
+          stroke: 'rgba(211,84,0 ,0.5)',
+          strokeWidth: 2,
+          angle: 210,
+          delay: 30,
+          speed: 0.2,
+          easing: mojs.easing.bezier(0.1, 1, 0.3, 1),
+          duration: tlDuration
+        }
+      })
+
+      const circleBurst = new mojs.Burst({
+        parent: '#clap',
+        radius: { 50: 75 },
+        angle: 25,
+        duration: tlDuration,
+        children: {
+          shape: 'circle',
+          fill: 'rgba(149,165,166 ,0.5)',
+          delay: 30,
+          speed: 0.2,
+          radius: { 3: 0 },
+          easing: mojs.easing.bezier(0.1, 1, 0.3, 1)
+        }
+      })
+
+      const countAnimation = new mojs.Html({
+        el: '#clapCount',
+        isShowStart: false,
+        isShowEnd: true,
+        y: { 0: -30 },
+        opacity: { 0: 1 },
+        duration: tlDuration
+      }).then({
+        opacity: { 1: 0 },
+        y: -80,
+        delay: tlDuration / 2
+      })
+
+      const countTotalAnimation = new mojs.Html({
+        el: '#clapCountTotal',
+        isShowStart: false,
+        isShowEnd: true,
+        opacity: { 0: 1 },
+        delay: (3 * tlDuration) / 2,
+        duration: tlDuration,
+        y: { 0: -3 }
+      })
+
+      const scaleButton = new mojs.Html({
+        el: '#clap',
+        duration: tlDuration,
+        scale: { 1.3: 1 },
+        easing: mojs.easing.out
+      })
+
+      const clap = document.getElementById('clap')
+      clap.style.transform = 'scale(1, 1)'
+      this.state.animationTimeline.add([
+        countAnimation,
+        countTotalAnimation,
+        scaleButton,
+        circleBurst,
+        triangleBurst
+      ])
+    }
+
+    render () {
+      return (
+        <WrappedComponent
+          animationTimeline={this.state.animationTimeline}
+          {...this.props}
+        />
+      )
+    }
+  }
+
+  WithClapAnimation.displayName = `WithClapAnimation(${getDisplayName(
+    WrappedComponent
+  )})`
+
+  return WithClapAnimation
+}
+
+function getDisplayName (WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
+}
+
+/** ====================================
+ *      🔰 MediumClap
+==================================== **/
+const initialState = {
+  count: 0,
+  countTotal: generateRandomNumber(500, 10000),
+  isClicked: false
+}
+
+const MediumClap = ({ animationTimeline }) => {
+  const MAXIMUM_USER_CLAP = 50
+  const [clapState, setClapState] = useState(initialState)
+  const { count, countTotal, isClicked } = clapState
+
+  const handleClapClick = () => {
+    // 👉 prop from HOC
+    animationTimeline.replay()
+
+    setClapState({
```

Comment:
🔥 IMPROVEMENT: 
Relying on the previous state when invoking the updater function returned from `useState`? It's better to pass a function to the updater e.g. 

```js
setClapState(prevState => ({
      count: Math.min(prevState.count + 1, MAXIMUM_USER_CLAP),
      countTotal: prevState.count < MAXIMUM_USER_CLAP ? prevState.countTotal + 1 : prevState.countTotal,
      isClicked: true
}))
```